client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/EqlDocumentationIT.java

```diff
@@ -0,0 +1,122 @@
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.client.WarningsHandler;
+import org.elasticsearch.client.eql.EqlSearchRequest;
+import org.elasticsearch.client.eql.EqlSearchResponse;
+import org.elasticsearch.client.eql.EqlSearchResponse.Event;
+import org.elasticsearch.client.eql.EqlSearchResponse.Hits;
+import org.elasticsearch.client.eql.EqlSearchResponse.Sequence;
+import org.elasticsearch.client.indices.CreateIndexRequest;
+import org.elasticsearch.client.indices.CreateIndexResponse;
+import org.elasticsearch.common.document.DocumentField;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.search.fetch.subphase.FieldAndFormat;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.xcontent.XContentType.JSON;
+
+/**
+ * Documentation for EQL APIs in the high level java client.
+ * Code wrapped in {@code tag} and {@code end} tags is included in the docs.
+ */
+@SuppressWarnings("removal")
+public class EqlDocumentationIT extends ESRestHighLevelClientTestCase {
+
+    @Before
+    void setUpIndex() throws IOException {
+        String index = "my-index";
+        CreateIndexResponse createIndexResponse = highLevelClient().indices().create(new CreateIndexRequest(index), RequestOptions.DEFAULT);
+        assertTrue(createIndexResponse.isAcknowledged());
+        BulkRequest bulk = new BulkRequest(index).setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+        bulk.add(new IndexRequest().source(JSON, "event_category", "process", "timestamp", "2021-11-23T00:00:00Z", "tie", 1, "host", "A"));
+        bulk.add(new IndexRequest().source(JSON, "event_category", "process", "timestamp", "2021-11-23T00:00:00Z", "tie", 2, "host", "B"));
+        bulk.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+        highLevelClient().bulk(bulk, RequestOptions.DEFAULT);
+    }
+
+    public void testEqlSearch() throws Exception {
+        RestHighLevelClient client = highLevelClient();
+
+        // tag::eql-search-request
+        String indices = "my-index"; // <1>
+        String query = "any where true"; // <2>
+        EqlSearchRequest request = new EqlSearchRequest(indices, query);
+        // end::eql-search-request
+
+        // tag::eql-search-request-arguments
+        request.eventCategoryField("event_category"); // <1>
+        request.fetchSize(50); // <2>
+        request.size(15); // <3>
+        request.tiebreakerField("tie"); // <4>
+        request.timestampField("timestamp"); // <5>
+        request.filter(QueryBuilders.matchAllQuery()); // <6>
+        request.resultPosition("head"); // <7>
+
+        List<FieldAndFormat> fields = new ArrayList<>();
+        fields.add(new FieldAndFormat("hostname", null));
+        request.fetchFields(fields); // <8>
```

Comment:
I think we need to avoid the `.of` usage pattern for anything that's not 8.x, since all the previous ES versions should be compatible with 1.8 as well. And this PR targets 7.x versions only.